crates/rpc/eth/block.rs

```diff
@@ -235,7 +246,76 @@ impl RpcHandler for GetRawHeaderRequest {
     }
 }
 
-pub fn get_all_block_receipts(
+impl RpcHandler for GetRawBlockRequest {
+    fn parse(params: &Option<Vec<Value>>) -> Result<GetRawBlockRequest, RpcErr> {
+        let params = params.as_ref().ok_or(RpcErr::BadParams)?;
+        if params.len() != 1 {
+            return Err(RpcErr::BadParams);
+        };
+
+        let block_str: String = serde_json::from_value(params[0].clone())?;
+        if !block_str.starts_with("0x") {
+            return Err(RpcErr::BadHexFormat);
+        }
```

Comment:
I find this confusing, the type of the block parameter is `BlockIdentifier`, meaning that it can be either "tag" such as "earliest", "latest", etc, or a number in hex format. With this code, the tag cases will be considered bad hex formats when they should be accepted.

Reply:
Updated!